### DISCUSSION TITLE
removing redundancies in search queries

### DISCUSSION THREAD

**Discussion by Hitcho84** [StractOrg/stract#244] [Q&A]
I'm constructing long search queries  joined with OR... There's a big chance that some of these terms are redundant because the search engine automatically searches for variations. Is there a way to know which search terms are redundant ? For example, for the search query "database" OR "list" OR  "Collection" OR "Repository"  OR " library", is there a way to shorten this query  by removing the redundant items? Or at least identifying the redundant ones and then I can remove them manually?

**Comment by mikkeldenker** [comment]
OR queries are actually not supported yet, so the query would just be treated as a large intersection query of the terms `['"database"', 'OR', '"list"', 'OR', '"Collection"', ...]` which is probably not what you want. The only way to get something like an OR query right now would be to search multiple times.

There is some deduplication of the terms so we try to limit the number of term lookups in each field. For instance, if we have the query `A A A A A` we would only lookup the term `A` once e.g. in the title field. It's a bit more complicated for ngram fields (for trigrams we would need to lookup, `A`, `A A`, and `A A A`) but overall the query planner tries to deduplicate the terms as much as possible beforehand.
